account_invoice_change_currency/models/account_change_currency.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2017-2018 Vauxoo
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import fields, models, api, _
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def action_account_change_currency(self):
+        track = self.env['mail.tracking.value']
+        for invoice in self.filtered(lambda x: x.state == 'draft'):
+            old_currency_id = invoice.get_last_currency_id()
+            if not old_currency_id or old_currency_id == invoice.currency_id:
+                continue
+            currency = invoice.with_context(
+                date=invoice.date_invoice or fields.Date.today()).currency_id
+            tracking_value_ids = [
+                [0, 0, track.create_tracking_values(
+                    currency, currency, 'currency_id',
+                    self.fields_get(['currency_id'])['currency_id'], 100)],
+                [0, 0, track.create_tracking_values(
+                    currency.rate, currency.rate,
+                    'rate',
+                    currency.fields_get(['rate'])['rate'], 100)],
+            ]
+            self.message_post(
+                body=_("Currency updated"),
```

Comment:
Is not the body extracted from the subtype?

In order to avoid add twice times this message.
<img width="354" alt="screen shot 2018-11-06 at 20 28 09" src="https://user-images.githubusercontent.com/6644187/48107378-01efb280-e205-11e8-9144-f533c4dd01bf.png">